src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -609,7 +621,20 @@ private Task ProcessRstStreamFrameAsync()
             }
 
             ThrowIfIncomingFrameSentToIdleStream();
-            AbortStream(_incomingFrame.StreamId, new IOException(CoreStrings.Http2StreamResetByClient));
+
+            if (_streams.TryGetValue(_incomingFrame.StreamId, out var stream))
+            {
+                // Second reset?
```

Comment:
Remove the question mark since we're sure this is could only be caused by a second reset.